From: Jan-Ivar Bruaroey <wingboy@gmail.com>
Date: Fri, 23 Jun 2023 11:09:36 -0400
Subject: (tmp-cherry-pick) Revert "Remove ISAC media constant and payload type
 mapping" (b79b74e08b)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This reverts commit 4c7271aafef89f62381f502f094e2a30421b2498.

Reason for revert: Breaks downstream test

Original change's description:
> Remove ISAC media constant and payload type mapping
>
> following the removal of ISAC from the code base.
>
> BUG=webrtc:14450
>
> Change-Id: I6faab5391bf0ef563c5dcce0bd5d8a653a87d9c8
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/294523
> Reviewed-by: Alessio Bazzica <alessiob@webrtc.org>
> Reviewed-by: Harald Alvestrand <hta@webrtc.org>
> Commit-Queue: Philipp Hancke <phancke@microsoft.com>
> Cr-Commit-Position: refs/heads/main@{#39378}

Bug: webrtc:14450
Change-Id: Idccd0ad7a05828f1be6db2071878c64d9bd37f33
No-Presubmit: true
No-Tree-Checks: true
No-Try: true
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/294742
Auto-Submit: Björn Terelius <terelius@webrtc.org>
Commit-Queue: Björn Terelius <terelius@webrtc.org>
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Owners-Override: Björn Terelius <terelius@webrtc.org>
Cr-Commit-Position: refs/heads/main@{#39380}
---
 media/base/media_constants.cc                        | 1 +
 media/base/media_constants.h                         | 1 +
 media/engine/payload_type_mapper.cc                  | 2 ++
 media/engine/payload_type_mapper_unittest.cc         | 2 ++
 sdk/objc/api/peerconnection/RTCRtpCodecParameters.mm | 1 +
 5 files changed, 7 insertions(+)

diff --git a/media/base/media_constants.cc b/media/base/media_constants.cc
index f91ac33c90..2f29a2036b 100644
--- a/media/base/media_constants.cc
+++ b/media/base/media_constants.cc
@@ -44,6 +44,7 @@ const char kCodecParamAssociatedCodecName[] = "acn";
 const char kCodecParamNotInNameValueFormat[] = "";
 
 const char kOpusCodecName[] = "opus";
+const char kIsacCodecName[] = "ISAC";
 const char kL16CodecName[] = "L16";
 const char kG722CodecName[] = "G722";
 const char kIlbcCodecName[] = "ILBC";
diff --git a/media/base/media_constants.h b/media/base/media_constants.h
index ca3213ea0f..f843d50ce5 100644
--- a/media/base/media_constants.h
+++ b/media/base/media_constants.h
@@ -45,6 +45,7 @@ extern const char kCodecParamAssociatedCodecName[];
 extern const char kCodecParamNotInNameValueFormat[];
 
 extern const char kOpusCodecName[];
+extern const char kIsacCodecName[];
 extern const char kL16CodecName[];
 extern const char kG722CodecName[];
 extern const char kIlbcCodecName[];
diff --git a/media/engine/payload_type_mapper.cc b/media/engine/payload_type_mapper.cc
index 66c48a776b..c63d1d7221 100644
--- a/media/engine/payload_type_mapper.cc
+++ b/media/engine/payload_type_mapper.cc
@@ -62,6 +62,8 @@ PayloadTypeMapper::PayloadTypeMapper()
            // Payload type assignments currently used by WebRTC.
            // Includes data to reduce collisions (and thus reassignments)
            {{kIlbcCodecName, 8000, 1}, 102},
+           {{kIsacCodecName, 16000, 1}, 103},
+           {{kIsacCodecName, 32000, 1}, 104},
            {{kCnCodecName, 16000, 1}, 105},
            {{kCnCodecName, 32000, 1}, 106},
            {{kOpusCodecName,
diff --git a/media/engine/payload_type_mapper_unittest.cc b/media/engine/payload_type_mapper_unittest.cc
index 92253a0f5d..90e113c7b6 100644
--- a/media/engine/payload_type_mapper_unittest.cc
+++ b/media/engine/payload_type_mapper_unittest.cc
@@ -50,6 +50,8 @@ TEST_F(PayloadTypeMapperTest, WebRTCPayloadTypes) {
   // Tests that the payload mapper knows about the audio formats we've
   // been using in WebRTC, with their hard coded values.
   EXPECT_EQ(102, mapper_.FindMappingFor({kIlbcCodecName, 8000, 1}));
+  EXPECT_EQ(103, mapper_.FindMappingFor({kIsacCodecName, 16000, 1}));
+  EXPECT_EQ(104, mapper_.FindMappingFor({kIsacCodecName, 32000, 1}));
   EXPECT_EQ(105, mapper_.FindMappingFor({kCnCodecName, 16000, 1}));
   EXPECT_EQ(106, mapper_.FindMappingFor({kCnCodecName, 32000, 1}));
   EXPECT_EQ(111, mapper_.FindMappingFor(
diff --git a/sdk/objc/api/peerconnection/RTCRtpCodecParameters.mm b/sdk/objc/api/peerconnection/RTCRtpCodecParameters.mm
index 6201e57b93..753667b635 100644
--- a/sdk/objc/api/peerconnection/RTCRtpCodecParameters.mm
+++ b/sdk/objc/api/peerconnection/RTCRtpCodecParameters.mm
@@ -21,6 +21,7 @@
 const NSString * const kRTCUlpfecCodecName = @(cricket::kUlpfecCodecName);
 const NSString * const kRTCFlexfecCodecName = @(cricket::kFlexfecCodecName);
 const NSString * const kRTCOpusCodecName = @(cricket::kOpusCodecName);
+const NSString * const kRTCIsacCodecName = @(cricket::kIsacCodecName);
 const NSString * const kRTCL16CodecName  = @(cricket::kL16CodecName);
 const NSString * const kRTCG722CodecName = @(cricket::kG722CodecName);
 const NSString * const kRTCIlbcCodecName = @(cricket::kIlbcCodecName);
-- 
2.37.1 (Apple Git-137.1)

